x/merkledb/db.go

```diff
@@ -163,6 +163,11 @@ type Config struct {
 	// BranchFactor determines the number of children each node can have.
 	BranchFactor BranchFactor
 
+	// Hasher defines the hash function to use when hashing the trie.
+	//
+	// If not specified, [SHA256Hasher] will be used.
```

Comment:
NIt: we could add a `var DefaultHasher = SHA256Hasher`. I think it might be a bit cleaner than referencing `SHA256Hasher` everywhere but I don't feel strongly.